.github/workflows/run-forge.yaml

```diff
@@ -0,0 +1,78 @@
+name: Run Forge
+
+on:
+  # this is called from within the build-images.yaml workflow since we depend on the images having been built before
+  workflow_call:
+    inputs:
+      GIT_SHA:
+        required: true
+        type: string
+        description:
+
+env:
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  AWS_REGION: us-west-2
+  IMAGE_TAG: ${{ inputs.GIT_SHA }}
+
+jobs:
+  forge:
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.GIT_SHA }}
+
+      - name: Ensure image exists
+        run: aws ecr describe-images --repository-name="aptos/validator" --image-ids=imageTag=$IMAGE_TAG
+      - name: Run Forge
+        shell: bash
+        run: |
+          set +e
+          FGI_REPRO_CMD="./scripts/fgi/run --tag $IMAGE_TAG --suite land_blocking --report forge_report.json"
+          eval $FGI_REPRO_CMD
+          FGI_EXIT_CODE=$?
+          echo "FGI_REPRO_CMD='$FGI_REPRO_CMD'" >> $GITHUB_ENV
+          echo "FGI_EXIT_CODE=$FGI_EXIT_CODE" >> $GITHUB_ENV
+
+      - name: Post Forge test results
+        shell: bash
+        env:
+          PUSH_GATEWAY: ${{ secrets.PUSH_GATEWAY }}
+          PUSH_GATEWAY_USER: ${{ secrets.PUSH_GATEWAY_USER }}
+          PUSH_GATEWAY_PASSWORD: ${{ secrets.PUSH_GATEWAY_PASSWORD }}
+          PR_NUMBER: ${{ github.event.number }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          set -x
+          
+          commit_message=$(git log -1 --pretty=%B)
+          PR_NUMBER="${PR_NUMBER:-$(echo "${commit_message}" | grep 'Closes: #' | tail -1 | sed 's/Closes: #//')}" 
+
+          echo "Repro: $FGI_REPRO_CMD"
+          FORGE_REPORT_TXT=$(cat forge_report.json | jq -r .text)
+          echo "FORGE_REPORT_TXT='$FORGE_REPORT_TXT'" >> $GITHUB_ENV
+
+          echo "Forge run: $GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID" >> forge_comment.txt
+          echo "Forge test result: $FORGE_REPORT_TXT" >> forge_comment.txt
+
+          # replace all newlines
+          FORGE_COMMENT=$(awk '{printf "%s\\n", $0}' forge_comment.txt)
+
+          # post  comment on the PR
```

Comment:
now that we're using GHA, can we do this in an action step? maybe https://github.com/marketplace/actions/add-pr-comment

Reply:
done